main.go

```diff
@@ -80,5 +92,49 @@ func (cli *CLI) create(args ...string) {
 func (cli *CLI) serve(args ...string) {
 	log.Printf("Shopify CLI Extensions Server is now available at http://localhost:%d/", cli.config.Port)
 	api := api.New(cli.config, ctx)
+
+	var develop_workers int
+	var watch_workers int
+
+	develop_chan := make(chan build.Result)
+	watch_chan := make(chan build.Result)
+
+	for _, e := range cli.config.Extensions {
+		b := build.NewBuilder(e)
+
+		go b.Develop(ctx, func(result build.Result) {
+			develop_chan <- result
+		})
+
+		go cli.monitor(develop_workers, develop_chan, "Develop", api)
+
+		go b.Watch(ctx, func(result build.Result) {
+			watch_chan <- result
+		})
+
+		go cli.monitor(watch_workers, watch_chan, "Watch", api)
+	}
+
 	go api.Start(ctx)
+
+	<-develop_chan
+	<-watch_chan
+}
+
+func (cli *CLI) monitor(active_workers int, ch chan build.Result, action string, a *api.ExtensionsApi) {
+	for result := range ch {
+		active_workers--
+
+		if result.Success {
+			log.Printf("[%s] event for extension: %s", action, result.UUID)
+			go a.Notify(api.StatusUpdate{Type: "Success", Extensions: cli.config.Extensions})
```

Comment:
Can we notify only with the updated extension? This way the Dev Console can update the status for the right extension

Reply:
Done! ✅